infra/build/functions/build_and_run_coverage.py

```diff
@@ -223,6 +233,71 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-arguments
       build_lib.http_upload_step(latest_report_info_body,
                                  latest_report_info_url,
                                  LATEST_REPORT_INFO_CONTENT_TYPE))
+
+  if project.fuzzing_language in LANGUAGES_WITH_INTROSPECTOR_SUPPORT:
+    build_steps.extend(
+        get_fuzz_introspector_steps(project, base_images_project, config,
+                                    bucket.html_report_url))
+  return build_steps
+
+
+def get_fuzz_introspector_steps(project, base_images_project, config,
+                                coverage_url):
+  """Return build steps of fuzz introspector for project"""
+  build_steps = []
+  build = build_project.Build(FUZZING_ENGINE, 'introspector', ARCHITECTURE)
+  env = build_project.get_env(project.fuzzing_language, build)
+
+  report_date = build_project.get_datetime_now().strftime('%Y%m%d')
+  bucket = IntrospectorBucket(project.name, report_date, PLATFORM,
+                              config.testing)
+
+  build_steps.append({
+      'name':
+          build_project.get_runner_image_name(base_images_project,
+                                              config.test_image_suffix),
+      'args': [
+          'bash', '-c',
+          ('sed -i s/base-builder/base-builder:introspector/g '
+           f'oss-fuzz/projects/{project.name}/Dockerfile')
+      ]
+  })
+
+  build_steps.append({
```

Comment:
Taking a look at cloud build, it doesn't seem like it supports an option to ignore build step failures. 

We could do this ourselves with `|| true` in every step here, but that's rather ugly. Could we just separate out the fuzz introspector steps into a separate build? We don't have to re-build the coverage build and generate the coverage reports. To start with, we can just use the latest coverage report at the time of requesting the build (i.e. most likely yesterday's) and pass it to the fuzz introspector build.

Reply:
Besides the coverage report, introspector relies on coverage data for each fuzz target generated [here](https://github.com/google/oss-fuzz/blob/23203939fc3032777f4440ccfeb4c8e19e7891d0/infra/base-images/base-runner/coverage#L116).
If we are to separate the two builds then we have two choices:

1) Re-use the coverage reports (as you mentioned with a ~1 day delay). This involves re-using the coverage report url, and for each `fuzz_target`, downloading `fuzz_target.covreport` generate by `coverage` script into introspector build.

2) Accept the overhead of re-running coverage steps in introspector build and avoid downloading steps and the ~1 day delay in coverage report. The compile step for introspector is the dominating step in terms of running time.

Should we still take approach 1? I personally prefer 2, because then introspector reports are as fresh as coverage reports.

Reply:
We can still start with 1.

As a next step, we can make the scheduling smarter such that in most cases the introspector will always use the latest coverage report and data rather than the 1 day old report.